staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1beta1/types.go

```diff
@@ -323,7 +323,7 @@ type ClaimMappings struct {
 	// If username.expression uses 'claims.email', then 'claims.email_verified' must be used in
 	// username.expression or extra[*].valueExpression or claimValidationRules[*].expression.
 	// An example claim validation rule expression that matches the validation automatically
-	// applied when username.claim is set to 'email' is 'claims.?email_verified.orValue(true)'.
+	// applied when username.claim is set to 'email' is 'claims.?email_verified.orValue(true) || false'.
```

Comment:
I don't understand what scenario this change is addressing... and absent `email_verified` claim should produce `true`... a present boolean `email_verified` claim should produce the value

Reply:
`orValue(true) || false` is a really strange construction

Reply:
The static compilation of the expression needs to always return a bool.  The value in claims doesn't have a known type.  The `|| false` statically guarantees a bool return value (or an error if the value in claims is not a bool).

Reply:
oh, we're trying to work around shortcomings in our return-type detection and use-of-email_verified detection ... both of those seem like bugs to fix, not just document our way around

Reply:
I don't really want to push the checks to runtime since we will lose the already minimal static validation we have on the expressions.

Reply:
`bool(claims.?email_verified.orValue(true))` seems to work as well and is a lot clearer

Reply:
these were the other options we were considering
- `bool(claims.?email_verified.orValue(true))`
- `claims.?email_verified.orValue(true) ? true : false`

Decided to go with `claims.?email_verified.orValue(true) || false` because it resembled the go code.

Reply:
`bool(...)` will convert a string to a bool as well, which doesn't match the existing Go code and in general probably isn't what we want.

Reply:
what about `claims.?email_verified.orValue(true) == true`?

Reply:
`claims.?email_verified.orValue(true) == true` does work and the check is explicit. @enj you good with this?

Reply:
I like `== true` the best out of the possible options.

Reply:
updated!